### PR TITLE
Add support for companion object parents

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Comment by dwijnand** [comment]
Hold on, something might be wrong.

File: core/src/main/scala_2.10/macrocompat/bundle_2.10.scala [hunk 1]
```diff
@@ -164,18 +164,23 @@ class BundleMacro[C <: Context](val c: C) {
...
         c.abort(c.enclosingPosition, "Unexpected tree shape.")
     }
   }
 
-  def mkMacroClsAndObjTree(clsDef: ClassDef, objBody: List[Tree]) = {
+  def mkMacroClsAndObjTree(clsDef: ClassDef, objDef: Option[ModuleDef]) = {
     val ClassDef(_, macroClassNme, _, Template(parents, self, body)) = clsDef
 
+    val (objEarlydefns, objParents, objBody) = objDef match {
+      case q"$objMods object $objTname extends { ..$objEarlydefns } with ..$objParents { $objSelf => ..$objBody }" => (objEarlydefns, objParents, objBody)
```
Comment (dwijnand) [inline core/src/main/scala_2.10/macrocompat/bundle_2.10.scala]: perhaps you want to match against Some, neh?

**Comment by dwijnand** [comment]
Ok, found the bug, added tests, and pushed. Ready for review.

**Comment by milessabin** [comment]
LGTM! Many thanks ... merging :-)

**Pull request merged**